paddle/phi/kernels/gpu/multi_tensor_adam_kernel.cu

```diff
@@ -0,0 +1,278 @@
+
+    for (int i_start = 0; i_start < n && i_start < compute_group_size;
+         i_start += blockDim.x * PARALLEL_SIZE) {
+      MT r_g[PARALLEL_SIZE];
+      MT r_p[PARALLEL_SIZE];
+      MT r_m[PARALLEL_SIZE];
+      MT r_v[PARALLEL_SIZE];
+#pragma unroll
+      for (int ii = 0; ii < PARALLEL_SIZE; ii++) {
+        int i = i_start + threadIdx.x + ii * blockDim.x;
+        if (i < n && i < compute_group_size) {
+          r_g[ii] = static_cast<MT>(g[i]);
+          r_p[ii] = multi_precision ? mp[i] : static_cast<MT>(p[i]);
+          r_m[ii] = static_cast<MT>(m[i]);
+          r_v[ii] = static_cast<MT>(v[i]);
+        } else {
+          r_g[ii] = MT(0);
+          r_p[ii] = MT(0);
+          r_m[ii] = MT(0);
+          r_v[ii] = MT(0);
+        }
+      }
+#pragma unroll
+      for (int ii = 0; ii < PARALLEL_SIZE; ii++) {
+        MT p = r_p[ii];
+        MT g = r_g[ii];
+        MT m = r_m[ii];
+        MT v = r_v[ii];
+        if (mode == false) {
+          m = beta1 * m + (static_cast<MT>(1.0) - beta1) * g;
+          v = beta2 * v + (static_cast<MT>(1.0) - beta2) * g * g;
+          r_m[ii] = m;
+          r_v[ii] = v;
+          MT denom =
+              (sqrt(v) / sqrt(static_cast<MT>(1.0) - beta2_pow)) + epsilon;
+          p += (m / denom) * (-(lr / (static_cast<MT>(1.0) - beta1_pow)));
+          r_p[ii] = p;
+        } else {  // weight decay
+          p *= (static_cast<MT>(1.0) - lr * decay);
+          m = beta1 * m + (static_cast<MT>(1.0) - beta1) * g;
+          v = beta2 * v + (static_cast<MT>(1.0) - beta2) * g * g;
+          r_m[ii] = m;
+          r_v[ii] = v;
+          MT denom =
+              (sqrt(v) / sqrt(static_cast<MT>(1.0) - beta2_pow)) + epsilon;
+          p += (m / denom) * (-(lr / (static_cast<MT>(1.0) - beta1_pow)));
+          r_p[ii] = p;
+        }
+      }
+#pragma unroll
+      for (int ii = 0; ii < PARALLEL_SIZE; ii++) {
+        int i = i_start + threadIdx.x + ii * blockDim.x;
+        if (i < n && i < compute_group_size) {
+          p[i] = static_cast<T>(r_p[ii]);
+          m[i] = r_m[ii];
+          v[i] = r_v[ii];
+          if (multi_precision) {
+            mp[i] = r_p[ii];
+          }
+        }
+      }
+    }
+  }
+};
+
+template <typename T>
+__global__ void UpdateBetaPow(T beta1,
+                              T beta2,
+                              const T* beta1_pow_,
+                              const T* beta2_pow_,
+                              T* beta1_pow_out,
+                              T* beta2_pow_out) {
+  *beta1_pow_out = beta1 * beta1_pow_[0];
+  *beta2_pow_out = beta2 * beta2_pow_[0];
+}
+
+template <typename T, typename Context>
+void MultiTensorAdamKernel(
+    const Context& dev_ctx,
+    const std::vector<const DenseTensor*>& params,
+    const std::vector<const DenseTensor*>& grads,
+    const std::vector<const DenseTensor*>& moments1,
+    const std::vector<const DenseTensor*>& moments2,
+    const paddle::optional<std::vector<const DenseTensor*>>& master_param,
```

Comment:
`master_param` -> `master_params`. Same below. And please fix `mode`, `master_param_out`, `compute_group_size`, etc.